src/app/work-item/work-item.service.ts

```diff
@@ -54,26 +66,35 @@ export class WorkItemService {
       .get(url, { headers: this.headers })
       .toPromise()
       .then(response => {
+        // Build the user - id map
         this.buildUserIdMap();
+        let wItems: WorkItem[];
         if (process.env.ENV == 'inmemory') {
           // Exclusively for in memory testing
-          this.workItems = response.json().data as WorkItem[];
-          this.workItems = this.workItems.reverse();
+          wItems = (response.json().data as WorkItem[]).reverse();
         } else {
           let links = response.json().links;
           if (links.hasOwnProperty('next')) {
             this.nextLink = links.next;
           }
-          this.workItems = response.json().data as WorkItem[];
+          wItems = response.json().data as WorkItem[];
         }
-        this.workItems.forEach((item) => {
+        wItems.forEach((item) => {
+          // Resolve the assignee and creator
           this.resolveUsersForWorkItem(item);
         });
+        // Update the existing workItem big list with new data
+        this.updateWorkItem(wItems);
         return this.workItems;
       })
       .catch (this.handleError); 
   }
 
+  /**
+   * This function is called from next page onwards in the scroll
+   * It does pretty much same as the getWorkItems function
+   */
+
```

Comment:
It is just a style thing but usually the JSDoc that goes with a function or class is directly above the function or class with no space or blank line. This visually ties them together.